### PR TITLE
Modify S6710(FCM): Improve regex, name and pattern

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**
The name should be plural because "should not be disclosed" is appended after

### CODE REVIEW

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/firebase.yaml [hunk 1]
```diff
@@ -16,11 +16,19 @@ provider:
...
       metadata:
-        name: Legacy Firebase Cloud Messaging API Key
+        name: Legacy Firebase Cloud Messaging API Keys
       detection:
         matching:
-          pattern: "(?i)\\b([a-z0-9\\-_]{11}:apa91b[a-z0-9\\-_+]{134})\\b"
+          pattern: "(?i)\\b([\\w\\-]{11}:apa91b[\\w\\-+]{134})\\b"
       examples:
+        - text: |
+            # Noncompliant code example
```
Comment (daniel-teuchert-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/firebase.yaml:25]: The noncompliant code example should raise an issue. Maybe the example needs to be adjusted?

**Review by daniel-teuchert-sonarsource — CHANGES_REQUESTED** [review]
The noncompliant example does not raise an issue.

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/firebase.yaml [hunk 2]
```diff
@@ -16,11 +16,19 @@ provider:
...
+            props.set("fcm_key", "APA91bJxU9oMf3..._...WoviAdUK5i-SB7iHcb4Wd38EMs...R9gIQyzpjoThmqe")
```
Comment (loris-s-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/firebase.yaml:26]:
```suggestion
            props.set("fcm_key", "APA91bJxU9oMf3RbiyqnmUO60KU_JLawjf2yrTfSs3_ZAp3dxZS0J88G5P5AoKWoviAdUK5i-2SB7iHcb4Wd38EMsZXBAAb6GZMaSOeKfaI0DuLxAFTOgGNKRSmj2R9gIQyzpjoThmqe")
```

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/firebase.yaml (hunk 1 above)
Comment (loris-s-sonarsource) [reply sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/firebase.yaml:25]: Done, and adjusted here: https://github.com/SonarSource/rspec/pull/3128

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/firebase.yaml [hunk 3]
```diff
@@ -16,11 +16,19 @@ provider:
...
+        name: Legacy Firebase Cloud Messaging API Keys
       detection:
         matching:
-          pattern: "(?i)\\b([a-z0-9\\-_]{11}:apa91b[a-z0-9\\-_+]{134})\\b"
+          pattern: "(?i)\\b([\\w\\-]{11}:apa91b[\\w\\-+]{134})\\b"
       examples:
+        - text: |
+            # Noncompliant code example
+            props.set("fcm_key", "APA91bJxU9oMf3RbiyqnmUO60KU_JLawjf2yrTfSs3_ZAp3dxZS0J88G5P5AoKWoviAdUK5i-2SB7iHcb4Wd38EMsZXBAAb6GZMaSOeKfaI0DuLxAFTOgGNKRSmj2R9gIQyzpjoThmqe")
+          containsSecret: false
```
Comment (daniel-teuchert-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/firebase.yaml:27]: This is still set to false. It should be true and the secret should be matched.

**Review by daniel-teuchert-sonarsource — CHANGES_REQUESTED** [review]
Noncompliant code example still does not match a secret.

**Review by daniel-teuchert-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**